### PR TITLE
Health doc updates

### LINKED ISSUE #6451: DOCS: 4.x Health Checks SE and MP
## Problem Description

We need to update docs for 4.x

- [x] update health.adoc for MP
- [x] update health.adoc for SE
- [x] update health.adoc guide for MP
- [x] update health.adoc guide for SE

## Guidelines

- [x] Make sure any fixes from SE 3.x are forward ported to 4.x https://github.com/helidon-io/helidon/pull/7515
- [x] Make sure any fixes from MP 3.x are forward ported to 4.x
- [x] Initial focus is on correctness
- [x] Record migration impact in this issue

## Migration Impact

Summarize incompatibilities between Helidon 3 and Helidon 4.
* MP: To set the path for the health endpoint, use the `health.endpoint` config key instead of `health.web-context`.
* SE: To set the path for the health endpoint:
   * Use the `features.observe.observers.health.endpoint` key instead of `health.web-context`. 
   * Use the `endpoint` method on the `HealthObserver.Builder` type instead of the `webContext` method on the `HealthSupport.Builder` type.
* SE and MP: To set up a non-default socket for health (and all observability endpoints) use the `server.features.observe.socket` config key:
   ```
   server:
     sockets:
       - name: "admin"
       - port: 8081
     features:
       observe:
         socket: "admin"
   ```
* SE: The `HealthSupport` type is replaced by the combination of `ObserveFeature` and `HealthObserver`. See the doc for details.
* SE: The default health endpoint is now at `/observe/health` instead of `/health` as in 3.x.
* SE: SE applications now use the `io.helidon.health.HealthCheck` and `...HealthCheckResponse` types instead of the counterpart types from MicroProfile Health.
* SE: Adding the `helidon-health` and `helidon-health-checks` dependencies to a project automatically adds the built-in health checks and the health endpoint to the application; application code no longer needs to explicitly add health to the application _unless_ it also disables `WebServer.Builder.featuresDiscoverServices`.

Summarize actions customers must take to migrate code
OK to reference external specs or documents (for example MicroProfile specs)

### PR DESCRIPTION
**Pull request by tjquinno**
### Description
Resolves #6451

### CODE REVIEW

File: docs/mp/health.adoc [hunk 1]
```diff
@@ -163,18 +164,18 @@ The example below will change the root path.
...
 == Examples
 
-Generate Helidon MP Quickstart project following these xref:guides/quickstart.adoc[Instruction]
+Generate Helidon MP Quickstart project following these xref:guides/quickstart.adoc[instructions].
 
 === Using the Built-In Health Checks
 
-Helidon has a set of built-in health checks that are enabled to report various
-health check statuses that are commonly used:
+Helidon has a set of built-in health checks which report various
```
Comment (ljamen) [inline docs/mp/health.adoc:177]: Helidon has a set of built-in health checks that can report various

File: docs/se/guides/health.adoc [hunk 2]
```diff
@@ -70,391 +69,199 @@ from the root directory of your project (helidon-quickstart-se).
 </dependency>
 ----
 
+Handling health checks is part of Helidon's observability support.
+By default, when you add the dependency for the built-in health checks, Helidon automatically registers the built-in checks.
+But the generated project does otherwise to illustrate how you can tailor its handling of health checks as you see fit.
```
Comment (ljamen) [inline docs/se/guides/health.adoc:74]: [question for clarity - not a change request] I read this to mean that the generated project does _not_ automatically register the built-in checks allowing you to configure the handling of health checks as needed.

**Review by ljamen — COMMENTED** [review]
small edits and a question (not a change request).

File: docs/se/guides/health.adoc (hunk 2 above)
Comment (tjquinno) [reply docs/se/guides/health.adoc:74]: Basically, yes.

I will change this phrasing to clarify what I'm trying to say. (I'm pushing another commit anyway to address the earlier comment.)

File: docs/mp/health.adoc (hunk 1 above)
Comment (tjquinno) [reply docs/mp/health.adoc]: Fixed.

**Review by ljamen — APPROVED** [review]
Changes incorporated- LGTM

**Pull request merged**